lock/etcd.go

```diff
@@ -60,46 +64,54 @@ func NewEtcdLockClient(keyapi KeysAPI, group string) (*EtcdLockClient, error) {
 
 	elc := &EtcdLockClient{keyapi, key}
 	if err := elc.Init(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to init etcd lock client: %w", err)
 	}
 
 	return elc, nil
 }
 
 // Init sets an initial copy of the semaphore if it doesn't exist yet.
+// So we first try to get the value, if the value is not found we create the key
+// with a default semaphore value
 func (c *EtcdLockClient) Init() error {
-	sem := newSemaphore()
-	b, err := json.Marshal(sem)
-	if err != nil {
-		return err
-	}
+	if _, err := c.Get(); err != nil {
```

Comment:
I'd flip the error logic here to save on indentation. `if err == nil { return nil }`, then `if !errors.Is(err, ErrNotFound) { return fmt.Errorf("...") }`, I think it should be easier to read.